Expensify/App
# Anthony 1542 issue pinned report

### Details
Skipping filtering of reports which have no messages but also are pinned

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/1542

### Tests - Unit
`npm run test`
I updated the test file OptionsListUtilsTest.js to count the number of pinned reports. One of the pinned reports doesn't have any messages. 

### Tests - Functional
1. Press the + button to start a new chat with an individual who you have never messaged before.
2. Before sending any message, pin the chat.
3. The chat should be pinned in the chat list on the left.
4. Click on another chat.
5. The chat should still pinned in the chat list on the left.
6. Click back on the chat and unpin the chat.
7. The chat should still be in the list but no longer be pinned.
8. Click on another chat
9. Chat should disappear from the chat list on the left.

### Tested On


As this is only logic in the data layer I didn't test this on other platforms. I'm relying on the unit tests and
